backend/analytics_server/tests/service/Incidents/test_mean_time_to_recovery.py

```diff
@@ -0,0 +1,137 @@
+import pytz
+from datetime import datetime
+from datetime import timedelta
+from tests.factories.models.incidents import get_mean_time_to_recovery_metrics
+from mhq.service.incidents.incidents import get_incident_service, IncidentService
+from mhq.utils.time import Interval
+
+from tests.factories.models import get_incident
+
+
+first_week_2024 = datetime(2024, 1, 1, 0, 0, 0, tzinfo=pytz.UTC)
+second_week_2024 = datetime(2024, 1, 8, 0, 0, 0, tzinfo=pytz.UTC)
+third_week_2024 = datetime(2024, 1, 15, 0, 0, 0, tzinfo=pytz.UTC)
+fourth_week_2024 = datetime(2024, 1, 22, 0, 0, 0, tzinfo=pytz.UTC)
+
+
+def test_get_incidents_mean_time_to_recovery_for_no_incidents():
+
+    incident_service = IncidentService(None, None)
+    incidents = []
+    mean_time_to_recovery = incident_service._get_incidents_mean_time_to_recovery([])
+
+    assert get_mean_time_to_recovery_metrics(None, 0) == mean_time_to_recovery
+
+
+def test_get_incidents_mean_time_to_recovery_for_incidents():
+
+    incident_service = IncidentService(None, None)
+
+    incident_1 = get_incident(
+        creation_date=first_week_2024,
+        resolved_date=first_week_2024 + timedelta(seconds=100),
+    )
+    incident_2 = get_incident(
+        creation_date=first_week_2024,
+        resolved_date=first_week_2024 + timedelta(seconds=300),
+    )
+    incident_3 = get_incident(
+        creation_date=first_week_2024,
+        resolved_date=first_week_2024 + timedelta(seconds=800),
+    )
+    incident_4 = get_incident(
+        creation_date=first_week_2024,
+        resolved_date=first_week_2024 + timedelta(seconds=90),
+    )
+
+    assert get_mean_time_to_recovery_metrics(
+        400, 3
+    ) == incident_service._get_incidents_mean_time_to_recovery(
+        [incident_1, incident_2, incident_3]
+    )
+
+    assert get_mean_time_to_recovery_metrics(
+        322.5, 4
```

Comment:
Numbers 400,3 and 322.5,4
I can't understand from test how/why these are being checked
What is logic behind 400/322.4.

if it is something you calculated from above incident fixtures then it would help if you could make them up adding them in code itself.

Reply:
it was the basic logic for incident time to restore / incident count. I have updated the code to make it more clear.